src/averageAges.js

```diff
@@ -55,10 +80,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  let children = people.filter(person => person.mother !== 0);
+
+  if (onlyWithSon) {
+    children = children.filter(person => person.sex === 'm');
+  }
+
+  children = children.map(person => {
+    const mother = people.find(isMother => isMother.name === person.mother);
+
+    if (mother) {
+      const ageDiff = (person.born - mother.born);
+
+      return ageDiff;
+    }
+  })
+    .filter(ageDiff => ageDiff !== undefined);
+
+  const countOfChildren = children.reduce((count) => (count + 1), 0);
+  const sumAgeDiff = children.reduce((sum, ageDiff) => (sum + ageDiff), 0);
+
+  return sumAgeDiff / countOfChildren;
```

Comment:
think here is also the same logic used in countAverageAge function, try to use it here to avoid code duplication